src/ohio/recipe/dbjoin.py

```diff
@@ -0,0 +1,117 @@
+"""
+dbjoin
+------
+
+Join the "COPY" results of arbitrary database queries in Python, without
+unnecessary memory overhead.
+
+This is largely useful to work around databases' per-query column limit.
+
+"""
+import contextlib
+import functools
+
+import ohio
+
+
+JOIN_QUERIES_SEP = ','
+JOIN_QUERIES_END = '\n'
+
+
+def db_join_queries(copy_sqls, engine, sep=JOIN_QUERIES_SEP, end=JOIN_QUERIES_END):
```

Comment:
I'm not sure if this function as written would ever be useful for other database systems. `copy_expert` is I think just a name psycopg2 came up with. Even Redshift, which can use the psycopg2 driver, wouldn't be implemented in this way (I think its copy needs to go straight to S3, no stdout available) Any high-throughput methods of data ingress/egress would have a different interface than a method named `copy_expert`, I think.

All of this to say I think you could just fold this function into pg_join_queries.

Reply:
Yep, I discuss that in the doc string.

I think it's useful to separate out concerns, especially for the sake of the recipe. If it's problematic as is, it could be given a leading underscore, and perhaps even hidden from the API. Or, it could be made more general, since the Python MySQL library _could_ in theory (be made to) do this. I just didn't want to worry about that prematurely.